presto-hive/src/test/java/com/facebook/presto/hive/TestStreamingAggregationPlan.java

```diff
@@ -0,0 +1,293 @@
+        }
+    }
+
+    @Test
+    public void testGroupbySuperset()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE ds_customer WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['custkey'], partitioned_by=array['ds'], \n" +
+                    "  format = 'DWRF' ) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM customer LIMIT 1000\n");
+
+            // can't enable stream
+            assertPlan(
+                    "SELECT custkey, name, COUNT(*) FROM ds_customer \n" +
+                            "WHERE ds = '2021-07-11' GROUP BY 1, 2",
+                    anyTree(aggregation(
+                            singleGroupingSet("custkey", "name"),
+                            // note: final aggregation function's parameter is of size one
+                            ImmutableMap.of(Optional.empty(), functionCall("count", false, ImmutableList.of(anySymbol()))),
+                            ImmutableList.of(), // non-streaming
+                            ImmutableMap.of(),
+                            Optional.empty(),
+                            FINAL,
+                            node(
+                                    ExchangeNode.class,
+                                    aggregation(
+                                            singleGroupingSet("custkey", "name"),
+                                            // note: partial aggregation function has no parameter
+                                            ImmutableMap.of(Optional.empty(), functionCall("count", ImmutableList.of())),
+                                            ImmutableList.of(), // non-streaming
+                                            ImmutableMap.of(),
+                                            Optional.empty(),
+                                            PARTIAL,
+                                            node(ProjectNode.class, tableScan("ds_customer", ImmutableMap.of("custkey", "custkey", "name", "name"))))))));
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS ds_customer");
+        }
+    }
+
+    @Test
+    public void testGroupbyMultiplekeys()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE ds_customer WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey', 'name'], \n" +
+                    "  sorted_by = ARRAY['custkey', 'name'], partitioned_by=array['ds'], \n" +
+                    "  format = 'DWRF' ) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM customer LIMIT 1000\n");
+
+            // can enable streaming aggregation
+            assertPlan(streamingAggregationEnabled(),
+                    "SELECT custkey, name, COUNT(*) FROM ds_customer \n" +
```

Comment:
Add another test with group by custkey(prefix of the bucket/sorted_by column). I think the streaming aggregation should be enabled.

Reply:
Yes, but that requires further changes in codebase which we're not proceeding right now, will add a todo here